packages/base-driver/lib/protocol/protocol.js

```diff
@@ -38,13 +34,28 @@ function extractProtocol (driver, sessionId = null) {
   }
 
   // Extract the protocol for the current session if the given driver is the umbrella one
-  return dstDriver ? dstDriver.protocol : SESSIONS_CACHE.getProtocol(sessionId);
+  return dstDriver?.protocol ?? PROTOCOLS.W3C;
 }
 
 function isSessionCommand (command) {
   return !_.includes(NO_SESSION_ID_COMMANDS, command);
 }
 
+function getLogger (driver, sessionId = null) {
```

Comment:
seems like we've defined this logger name generator logic in several places. should we consolidate it into one helper?

Reply:
not sure what exactly could be extracted into a common function here